Makefile

```
@@ -0,0 +1,94 @@
+# Shortcuts for various dev tasks. Based on makefile from pydantic
+.DEFAULT_GOAL := all
+isort = isort pytest_inmanta tests examples
+black = black pytest_inmanta tests examples
+
+.PHONY: install
+install:
+	pip install -U setuptools pip
+	pip install -U -r requirements.txt -r requirements.dev.txt
+	pip install -e .
+
+.PHONY: format
+format:
+	$(isort)
+	$(black)
+
+.PHONY: pep8
+pep8:
+	pip install -c requirements.txt pep8-naming flake8-black flake8-isort
+	flake8 src tests tests_common
+
+.PHONY: mypy mypy-diff mypy-save
+RUN_MYPY=MYPYPATH=stubs:src python -m mypy --soft-error-limit=-1 --html-report mypy -p pytest_inmanta
+
+mypy:
+	$(RUN_MYPY)
+
+# baseline file mypy-diff will compare to
+MYPY_BASELINE_FILE=.mypy-baseline
+# temporary file used to store most recent mypy run
+MYPY_TMP_FILE=.mypy-tmp
+# temporary file used to store baseline with line numbers filtered out
+MYPY_BASELINE_FILE_NO_LN_NB=$(MYPY_BASELINE_FILE).nolnnb
+# prepare file for diff: remove last 2 lines and filter out line numbers
+MYPY_DIFF_PREPARE=head -n -2 | sed 's/^\(.\+:\)[0-9]\+\(:\)/\1\2/'
+# read old/new line number (format +n for new or -n for old) from stdin and transform to old/new line
+MYPY_DIFF_FETCH_LINES=xargs -I{} sh -c 'sed -n -e "s/^/$(MYPY_SET_COLOUR)$$(echo {} | cut -c 1 -) /" -e "$$(echo {} | cut -c 2- -)p" $(MYPY_SELECT_FILE)'
```

```suggestion
MYPY_DIFF_FETCH_LINES=xargs -I{} bash -c 'sed -n -e "s/^/$(MYPY_SET_COLOUR)$$(echo {} | cut -c 1 -) /" -e "$$(echo {} | cut -c 2- -)p" $(MYPY_SELECT_FILE)'
```
As @FloLey found out this is not sh-compatible, it's actually bash syntax.